packages/services/src/services/DDO/DDO.js

```diff
@@ -0,0 +1,62 @@
+import root from 'window-or-global';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+
+/**
+ * DDO API class with methods of fetching search results for
+ * ibm.com
+ */
+class DDOAPI {
+  /**
+   * Promise function that determines when the digital data object is ready
+   *
+   * @returns {Promise} Resolved data layer ready signal
+   */
+  static isReady() {
+    return new Promise(resolve => {
+      if (
+        root.digitalData &&
+        root.digitalData.page &&
+        root.digitalData.page.isDataLayerReady
+      ) {
+        resolve();
+      } else {
+        root.addEventListener('datalayer_ready', () => resolve);
+      }
+    });
+  }
+
+  /**
+   * Gets the full digitalData (DDO) object
+   *
+   * @returns {Promise<*>} Promise object
+   */
+  static async getAll() {
```

Comment:
Should adopters be setting the DDO using this service moving forward?

Reply:
we should probably set up setters and getters for them for specific items I think. But if they need to set something specific, they can always use our `isReady` wrapper before they do it.